app/workers/functions.php

```diff
@@ -477,21 +475,26 @@ public function execute(string $trigger, string $projectId, string $executionId,
             ->setParam('eventData', $execution->getArrayCopy(array_keys($executionModel->getRules())));
 
         $executionUpdate->trigger();
-
-        $usage = new Event('v1-usage', 'UsageV1');
-
-        $usage
-            ->setParam('projectId', $projectId)
-            ->setParam('functionId', $function->getId())
-            ->setParam('functionExecution', 1)
-            ->setParam('functionStatus', $functionStatus)
-            ->setParam('functionExecutionTime', $executionTime * 1000) // ms
-            ->setParam('networkRequestSize', 0)
-            ->setParam('networkResponseSize', 0)
-        ;
         
         if(App::getEnv('_APP_USAGE_STATS', 'enabled') == 'enabled') {
-            $usage->trigger();
+            $statsd = $register->get('statsd');
+
+            $storage = 0;
+
+            $functionExecutionTime = $executionTime * 1000;
+
+            $tags = ",project={$projectId},version=".App::getEnv('_APP_VERSION', 'UNKNOWN');
+
+            // the global namespace is prepended to every key (optional)
+            $statsd->setNamespace('appwrite.usage');
+
+            $statsd->increment('executions.all'.$tags.',functionId='.$function->getId().',functionStatus='.$functionStatus);
+            $statsd->count('executions.time'.$tags.',functionId='.$function->getId(), $functionExecutionTime);
+
+            if($storage >= 1) {
```

Comment:
For what do we need this, if the condition will never trigger? 🙂

Reply:
something i missed :D